src/CTA.Rules.Common/WebConfigManagement/WebConfigManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Xml.Linq;
+using CTA.Rules.Config;
+
+namespace CTA.Rules.Common.WebConfigManagement
+{
+    /// <summary>
+    /// Consumers can use this class to load a Web.config into memory specify its representation based on need.
+    /// For accessing more well-defined Web.config properties (e.g. connection strings), it can be loaded as a
+    /// Configuration object. For more flexible navigation, it can be loaded as an XDocument with additional
+    /// helper functions.
+    /// </summary>
+    public class WebConfigManager
+    {
+        private static Dictionary<string, Configuration> _configurationCache = new Dictionary<string, Configuration>();
+        private static Dictionary<string, XDocument> _xDocumentCache = new Dictionary<string, XDocument>();
+        private delegate object ConfigLoadingDelegate(string configFile);
+
+        public static Configuration LoadWebConfigAsConfiguration(string projectDir)
+        {
+            var config = LoadWebConfig(projectDir, webConfigFile =>
+            {
+                if (_configurationCache.TryGetValue(webConfigFile, out var cached))
+                {
+                    return cached;
+                }
+
+                var fileMap = new ExeConfigurationFileMap {ExeConfigFilename = webConfigFile};
+                var configuration = ConfigurationManager.OpenMappedExeConfiguration(fileMap, ConfigurationUserLevel.None);
+                _configurationCache[webConfigFile] = configuration;
+
+                return _configurationCache[webConfigFile];
+            }) as Configuration;
+            
+            return config;
+        }
+
+        public static WebConfigXDocument LoadWebConfigAsXDocument(string projectDir)
+        {
+            var config = LoadWebConfig(projectDir, webConfigFile =>
+            {
+                if (_xDocumentCache.TryGetValue(webConfigFile, out var cached))
+                {
+                    return cached;
+                }
+
+                var xDocument = XDocument.Load(webConfigFile);
+                _xDocumentCache[webConfigFile] = xDocument;
+
+                return _xDocumentCache[webConfigFile];
+            }) as XDocument;
+
+            return new WebConfigXDocument(config);
+        }
+
+        public static void ClearCache()
+        {
+            _configurationCache = new Dictionary<string, Configuration>();
+            _xDocumentCache = new Dictionary<string, XDocument>();
```

Comment:
nit: Why not just set to null?

Reply:
The cache objects are static so if we set them to null and try loading another config, we'll get a null reference error. Open to other ideas if there's a better way to approach this.

Reply:
One way to do it can be to have a get where the object is initialized if it's null. This way we're not creating an object unless we need it.

Reply:
Sounds good, will add this